sql-plugin/src/main/scala/org/apache/spark/sql/catalyst/json/rapids/GpuJsonScan.scala

```diff
@@ -74,21 +78,38 @@ object GpuJsonScan {
       scan.dataSchema,
       scan.readDataSchema,
       scan.options.asScala.toMap,
-      scanMeta)
+      scanMeta,
+      scan.fileIndex.allFiles())
   }
 
   def tagSupport(
       sparkSession: SparkSession,
       dataSchema: StructType,
       readSchema: StructType,
       options: Map[String, String],
-      meta: RapidsMeta[_, _, _]): Unit = {
+      meta: RapidsMeta[_, _, _],
+      files: Seq[FileStatus]): Unit = {
 
     val parsedOptions = new JSONOptionsInRead(
       options,
       sparkSession.sessionState.conf.sessionLocalTimeZone,
       sparkSession.sessionState.conf.columnNameOfCorruptRecord)
 
+    // Since cudf (https://github.com/rapidsai/cudf/issues/9990) does not support
+    // column pruning for now, we just fallback to CPU.
+    val optSchema = JsonDataSource(parsedOptions).inferSchema(sparkSession, files, parsedOptions)
```

Comment:
This is crazy expensive to do. By default it will read all of the json files and not even sub-sample them. It will be as expensive as parsing the JSON on the CPU. It will never speed up processing. I would rather keep the bad user experience and document it instead.